File: app/src/main/java/com/lgvalle/material_animations/Sample.java
package com.lgvalle.material_animations;

import androidx.annotation.ColorInt;
import androidx.databinding.BindingAdapter;
import androidx.annotation.ColorRes;
import androidx.core.graphics.drawable.DrawableCompat;
import android.widget.ImageView;

import java.io.Serializable;

public class Sample implements Serializable {

    final int color;
    private final String name;

    public Sample(@ColorInt int color, String name) {
        this.color = color;
        this.name = name;
    }

    @BindingAdapter("bind:colorTint")
    public static void setColorTint(ImageView view, @ColorInt int color) {
        DrawableCompat.setTint(view.getDrawable(), color);
        //view.setColorFilter(color, PorterDuff.Mode.SRC_IN);
    }

    public String getName() {
        return name;
    }

    public int getColor() {
        return color;
    }


}

File: app/src/main/java/cn/woolsen/material/animations/Sample.kt
package cn.woolsen.material.animations

import android.widget.ImageView
import androidx.annotation.ColorInt
import androidx.databinding.BindingAdapter
import java.io.Serializable

class Sample(@ColorInt val color: Int, val name: String) : Serializable {

    companion object {
        @JvmStatic
        @BindingAdapter("colorTint")
        fun setColorTint(view: ImageView, @ColorInt color: Int) {
            view.setColorFilter(color)
        }
    }
}